app/assets/javascripts/lit/sources.js.coffee

```diff
@@ -0,0 +1,26 @@
+$ ->
+  sourceId = $('#source_id').attr('value')
+  apiKey = $('#api_key').attr('value')
+
+  updateFunc = ->
+    $.ajax "/lit/api/v1/sources/" + sourceId + "/sync_complete",
+      type: 'GET'
+      headers:
+        'Authorization': "Token token=\"#{apiKey}\""
+      success: (xml, textStatus, xhr) ->
+        if xhr.responseJSON.sync_complete
+          $('.loading').addClass('loaded').removeClass('loading')
+          clearInterval(interval)
+          location.reload()
+      statusCode:
+        404: ->
+          $('.loading').text('Could not update synchronization status, please try refreshing page')
+        401: ->
+          $('.loading').text('Invalid API key, please check source settings')
```

Comment:
I'm still not convinced :) But now I see this works differently than I thought at beginning. Have you tested it with two servers with two different api keys? You send @source.api_key as token, but in controller it is compared with `Lit.api_key` which has different value. You should send `Lit.api_key` in such case (I know we talked about this in person and looks like your previous code was correct).

But I don't think there is need to use `api_key` for this action at all. `sync_complete` should be just moved to `Lit::IncommingLocalizationsController`.


Once more what happen now (as I understand it):

S1 (`api_key: 's1'`) is a server when user is logged in and want to fetch translations from S2 (`api_key: 's2'`). He create source record with `api_key: 's2'`. When he opens `/lit/incomming_localizations` (S1) the ajax call is made to `/lit/api/v1/sources/" + sourceId + "/sync_complete` (S1) with `api_key: 's2'`.  He gets 401 error indeed, but he shouldn't in such case.